src/Mobs/Path.cpp

```diff
@@ -213,6 +207,76 @@ bool cPath::Step_Internal()
 
 
 
+void cPath::AttemptToFindAlternative()
+{
+	Vector3i PotentialNewDestination;
+	// Attempt to find a nearby path.
+	int OriginalX = (rand() % 5) - 2;
+	int OriginalZ = (rand() % 5) - 2;  // A random value between -2 and 2.
+	int x, y, z;
+	x = OriginalX;
+	z = OriginalZ;
+	// Scan a 5x3x5 cube around the original path, starting with a random spot.
+	// TODO: scan a 5x1x5 and If a block in the scan area is air, we scan the solid below instead.
+	for (;;)
+	{
+		for (;;)
+		{
+			for (y = -3; y <= 3; y+= 1)
+			{
+				if ((x == 0) && (y == 0) && (z == 0))
```

Comment:
Is it problematic to check the original destination? It seems that this only increases the complexity of the code unnecessarily.